torchtitan/experiments/llama4/train_configs/debug_model.toml

```
@@ -29,7 +29,9 @@ use_flex_attn = false
 attn_mask_type = "causal"  # causal / block_causal
 
 [optimizer]
-name = "AdamW"
+# TODO: currently grouped mm in MoE doesn't work with AdamW, need to investigate
+# name = "AdamW"
+name = "Adam"
```

```suggestion
name = "Adam"
fused=True
```

fused should be faster and more performant with memory.
I'd be curious if using fused instead of foreach (the default) would avoid the hang somehow.

---

re: I'd be curious if using fused instead of foreach (the default) would avoid the hang somehow.

atm we have fused default for AdamW already set in Titan:
https://github.com/pytorch/torchtitan/blob/981552ee0c5ed9bedf8c8b4967cf68c1becbf682/torchtitan/config_manager.py#L256